webapp/src/components/NavigateRegionsOverlay.tsx

```diff
@@ -0,0 +1,222 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import * as ReactDOM from "react-dom";
+import { Button, ButtonProps } from "../../../react-common/components/controls/Button";
+import { FocusTrap } from "../../../react-common/components/controls/FocusTrap";
+
+import IProjectView = pxt.editor.IProjectView;
+
+interface NavigateRegionsOverlayProps {
+    parent: IProjectView;
+}
+
+interface RectBounds {
+    top: number;
+    bottom: number;
+    left: number;
+    right: number;
+    width: number;
+    height: number;
+}
+
+type Region = "mainmenu" | "simulator" | "toolbox" | "workspace" | "editortools"
+
+const shortcutToRegion: Record<string, Region> = {
+    "1": "mainmenu",
+    "2": "simulator",
+    "3": "toolbox",
+    "4": "workspace",
+    "5": "editortools"
+}
+
+const regionToShortcut = Object.entries(shortcutToRegion).reduce((acc, entry) => {
+    const [shortcut, region] = entry;
+    acc[region] = shortcut;
+    return acc;
+  }, {} as Record<Region, string>);
+
+export const NavigateRegionsOverlay = ({ parent }: NavigateRegionsOverlayProps) => {
+    const isSimMini = () => !!document.querySelector(".miniSim")
+    const previouslyFocused = useRef<Element>(document.activeElement);
+    const getRects = () => ({
+        simulator: (
+            isSimMini()
+                ? document.querySelector(".simPanel")
+                : parent.state.collapseEditorTools ?
+                document.querySelector("#computertogglesim")
+                : document.querySelector("#editorSidebar")
+            ).getBoundingClientRect(),
+        toolbox:  document.querySelector(`${parent.isBlocksActive() ? ".blocklyToolbox" : ".monacoToolboxDiv"}`).getBoundingClientRect(),
+        workspace: document.querySelector(`${parent.isBlocksActive() ? ".blocklySvg" : ".monacoEditorArea"}`).getBoundingClientRect(),
+        mainmenu: document.querySelector("#mainmenu").getBoundingClientRect(),
+        editortools: document.querySelector("#editortools").getBoundingClientRect(),
+    })
+    const [regionRects, setRegionRects] = useState(getRects())
+
+    const focusRegion = useCallback((region: Region) => {
+        switch (region) {
+            case "mainmenu": {
+                (document.querySelector(".blocks-menuitem") as HTMLElement).focus();
+                parent.hideNavigateRegions();
+                return
```

Comment:
Maybe we break in the branches and do this at the end?
Requires corresponding change below to not call this with an undefined region if the `shortcutToRegion` lookup fails.

Reply:
Refactored here [8471d4f](https://github.com/microsoft/pxt/pull/10551/commits/8471d4fd946481fd450a89c71c3530ef28e471d8)